Include/object.h

```diff
@@ -283,6 +283,19 @@ _Py_ThreadId(void)
     // Both GCC and Clang have supported __builtin_thread_pointer
     // for s390 from long time ago.
     tid = (uintptr_t)__builtin_thread_pointer();
+#elif defined(thread_local) || defined(__GNUC__)
+    #if defined(thread_local)
+    static thread_local int __tp = 0;
+    #else
+    // Assume that we only support C11 compilers.
+    static __thread int __tp = 0;
+    #endif
+    // gh-112535: Using characteristics of TLS address mapping.
+    // The address of the thread-local variable is not equal with the actual thread pointer,
+    // However, it has the property of being determined by loader at runtime, with no duplication of values
```

Comment:
I am not sure it's proper explaination. Some values are initialzied at the startup and some values are blahblahblah..